File: COMP9041/lab07/total_words.pl
#!/usr/bin/perl -w

# count words from stdin
# all non [a-zA-Z] separate words will be regarded as separaters
$count = 0;
while ($line = <STDIN>){
    # remove all non-word chars at head, otherwise the length will be one more
    $line =~ s/^[^A-Za-z]+//;
    @splitted_line = split('[^A-Za-z]+', $line);
    $length = @splitted_line;
    $count = $count + $length;
}
print "$count words\n";

File: COMP9041/lab07/total_words.py
#!/usr/bin/env python

# count words from stdin
# all non [a-zA-Z] separate words will be regarded as separaters
import sys
import re

count = 0
for line in sys.stdin:
    pattern_head = re.compile("^[^A-Za-z]+")
    line = re.sub(pattern_head, '', line)
    pattern_split = re.compile("[^A-Za-z]+")
    splitted_line = re.split(pattern_split, line)
    count += len(splitted_line) - 1 # remove "\n"

print "{} words".format(count)
